cargo-near-build/src/near/build/mod.rs

```diff
@@ -50,7 +50,16 @@ pub fn run(args: Opts) -> eyre::Result<CompilationArtifact> {
 
     let out_dir = crate_metadata.resolve_output_dir(args.out_dir.map(Into::into))?;
 
-    let mut build_env = vec![("RUSTFLAGS", "-C link-arg=-s")];
+    let mut build_env;
+    if rustc_version::version().unwrap() >= rustc_version::Version::parse("1.82.0").unwrap() {
+        build_env = vec![(
+            "RUSTFLAGS",
+            "-C link-arg=-s -C target-feature=-multivalue,-reference-types",
+        )];
+    } else {
+        build_env = vec![("RUSTFLAGS", "-C link-arg=-s")];
+    }
```

Comment:
Just a syntax nitpick:

```suggestion
    let mut build_env = if rustc_version::version().unwrap() >= rustc_version::Version::parse("1.82.0").unwrap() {
        vec![(
            "RUSTFLAGS",
            "-C link-arg=-s -C target-feature=-multivalue,-reference-types",
        )]
    } else {
        vec![("RUSTFLAGS", "-C link-arg=-s")]
    }
```